python-package/setup.py

```diff
@@ -177,9 +165,41 @@ def compile_cpp(use_mingw=False, use_gpu=False, use_mpi=False, nomp=False,
                             error_msg='Please install CMake first')
     else:  # Linux, Darwin (macOS), etc.
         logger.info("Starting to compile with CMake.")
-        silent_call(cmake_cmd, raise_error=True, error_msg='Please install CMake and all required dependencies first')
-        silent_call(["make", "_lightgbm", "-j4"], raise_error=True,
-                    error_msg='An error has occurred while building lightgbm library file')
+        # Apple Clang with OpenMP
+        if system() == 'Darwin' and not nomp and not (os.environ.get('CC', '').startswith('gcc')
+                                                      and os.environ.get('CXX', '').startswith('g++')):
+            def get_cmake_opts(openmp_include_dir, openmp_library):
+                if openmp_include_dir and openmp_library:
+                    return ['-DOpenMP_C_FLAGS=-Xpreprocessor -fopenmp -I{0}'.format(openmp_include_dir),
+                            '-DOpenMP_C_LIB_NAMES=omp',
+                            '-DOpenMP_CXX_FLAGS=-Xpreprocessor -fopenmp -I{0}'.format(openmp_include_dir),
+                            '-DOpenMP_CXX_LIB_NAMES=omp',
+                            '-DOpenMP_omp_LIBRARY={0}'.format(openmp_library)]
+                else:
+                    return []
+
+            status = silent_call(cmake_cmd + get_cmake_opts(openmp_include_dir, openmp_library))
+            status += silent_call(["make", "_lightgbm", "-j4"])
+            if status != 0:
+                logger.warning("Compilation failed.")
+                logger.info("Starting to compile with Homebrew OpenMP paths guesses.")
+                clear_path(os.path.join(CURRENT_DIR, "build_cpp"))
+                status = silent_call(cmake_cmd + get_cmake_opts('/usr/local/opt/libomp/include',
+                                                                '/usr/local/opt/libomp/lib/libomp.dylib'))
+                status += silent_call(["make", "_lightgbm", "-j4"])
+            if status != 0:
+                logger.warning("Compilation failed.")
+                logger.info("Starting to compile with MacPorts OpenMP paths guesses.")
+                clear_path(os.path.join(CURRENT_DIR, "build_cpp"))
+                silent_call(cmake_cmd + get_cmake_opts('/opt/local/include/libomp',
+                                                       '/opt/local/lib/libomp/libomp.dylib'),
```

Comment:
@henry0312 
 >  MacPorts uses `/opt/local` as the default prefix.

Here it is.